src/coronawhy_vt/utils/local_processing.py

```diff
@@ -0,0 +1,375 @@
+    :param run_body: If true, features extracted from the body will be included in the aggregate
+    :param replace_dict (dict): {token: replacement_token}
+    """
+    if lemmatized:
+        pull_mentions = pull_mentions_lemmatized
+    else:
+        pull_mentions = pull_mentions_direct
+
+    data_dicts = []
+    if run_title:
+        data_dicts.append(pull_mentions(paths, title_blocks, "title", words, replace_dict))
+    if run_abstract:
+        data_dicts.append(pull_mentions(paths, abstract_blocks, "abstract", words, replace_dict))
+    if run_body:
+        data_dicts.append(pull_mentions(paths, body_blocks, "body", words, replace_dict))
+
+    summed_dictionary = data_dicts[0]
+    for k in data_dicts[0].keys():
+        for d in data_dicts:
+            summed_dictionary[k] = summed_dictionary[k] + d[k]
+
+    dat = pd.DataFrame(summed_dictionary)
+    dat.to_csv(file_name)
+
+
+# TODO change index to be of type int rather than string
+def same_sentence_check(block: str, pos1: str, pos2: str) -> bool:
+    """Check if two words are in the same sentence by looking for sentence delimiters between their starting positions.
+
+    :param block: Block of string text the two words are found in
+    :param pos1: The index of the beginning of word1
+    :param pos2: The index of the beginning of word2
+
+    :return: true if they word1 and word2 are not separated by
+        one of the follwing sentence delimiters ., ;, ?, !, false otherwise
+    """
+    if pos1 < pos2:
+        interstring = block[int(pos1):int(pos2)]
+    else:
+        interstring = block[int(pos2):int(pos1)]
+    sentence_enders = [".", ";", "?", "!"]
+    for s in sentence_enders:
+        if s in interstring:
+            return 0
+    return 1
+
+# This function makes the 2D quilt plot for showing co-occurences at block
+#   or sentence level of various classes of search terms
+#
+
+
+def make_2d_plot(dat_joined: pd.DataFrame, factor_1: str, factor_2: str, single_sentence_plots: bool = False) -> None:
+    """Create 2D quilt plot from dataframe row columns ('word_' + factor_1) and ('word_' + factor_2).
+
+    :param dat_joined: Dataframe of the format returned by pull_mentions_lemmatized and pull_mentions_direct
+    :param factor_1: x-axis of the graph, possible entries 'virus', 'therapy', 'drug', 'exp'
+    :param factor_2: y-axis of the graph, possible entries 'virus', 'therapy', 'drug', 'exp'
+    :param single_sentece_plots: If true, plot only coocurrences in the same sentence
+    """
+    if single_sentence_plots:
+        grouped = dat_joined[dat_joined.same_sentence is True].groupby(['word_' + factor_1, 'word_' + factor_2])
+    else:
+        grouped = dat_joined.groupby(['word_' + factor_1, 'word_' + factor_2])
+
+    values = grouped.count().values[:, 0]
+
+    index = grouped.count().index
+
+    # Holds aggregated, in order appearences of factor_1 and factor_2 respectively
+    index_1 = []
```

Comment:
there's a totally mindblowing pattern with `zip()` for this that lets you do it in one line:

```python
index_1, index_2 = zip(*index)
```